lib/shared/puppet_forge/module_name.rb

```diff
@@ -0,0 +1,23 @@
+module PuppetForge
+  # Normalize Forge module names
+  #
+  # Forge modules use '-' as the separator between the username and modulename
+  # in the module slug, but the Puppet Forge uses '/' as the separator on module
+  # pages and in the URL. This class hides the distinction by always using '-' as
+  # the separator for normal use. If the name needs to use the slash separated name
+  # then the #forge_name method can be used.
+  #
+  # @api private
+  class ModuleName < String
+
+    def initialize(*args)
+      super
+      tr!('/', '-')
+    end
+
+    # @return [String] The slash separated module name.
+    def forge_name
+      String.new(self).tr('-', '/')
+    end
```

Comment:
Since this subclasses string can't we just do ```self.tr('-', '/')``` here? 

But bigger picture, do we even need this method here? The fact that the Forge website uses "/" is really just an implementation detail that I'm not sure consumers of this library need to be concerned with. If we decide to keep it, something like "to_path" might be a more appropriately generic name for it.